src/databricks/labs/remorph/reconcile/connectors/data_source.py

```diff
@@ -25,13 +30,23 @@ def __init__(
         self.scope = scope
 
     @abstractmethod
-    def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOptions) -> DataFrame:
+    def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOptions | None) -> DataFrame:
         return NotImplemented
 
     @abstractmethod
     def get_schema(self, catalog: str, schema: str, table: str) -> list[Schema]:
         return NotImplemented
 
+    @abstractmethod
+    def list_tables(
+        self,
+        catalog: str,
+        schema: str,
+        include_list: list[str] | None,
+        exclude_list: list[str] | None,
+    ) -> TableRecon:
+        return NotImplemented
```

Comment:
is this expected to be covered in a different PR?

Reply:
Yes, will be 2 different PRs for Oracle and Snowflake